.github/workflows/test-action.yaml

```diff
@@ -2,11 +2,34 @@ name: Test Action
 
 on:
   workflow_dispatch:
+  push:
+    branches:
+      - main
   pull_request:
     branches:
       - main
 
 jobs:
+
+  automated-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
```

Comment:
Let's pin it to a nvmrc file in the root folder so that we gain from a future update without the need to change this string